### DISCUSSION TITLE
Shared memory in python

### DISCUSSION THREAD

**Discussion by iloveitaly** [capnproto/capnproto#1991] [Q&A]
Is shared memory RPC supported in the python library?

What I'd like to do is:

1. Have multiple python processes (I wish I could use threads, but I can't)
2. A child process finishes it's work and shares a massive data structure with a parent via cap-n-proto _without_ serializing it in any way (for performance)

Is this possible with python?

**Comment by kentonv** [comment]
At present it would be up to you to set up shared memory itself; capnp provides no particular helpers for this (even in C++). Once you have a shared memory segment, you could write a MessageBuilder implementation that writes into it, and a MessageReader on the other side that reads from it. I don't know if it's possible to write custom MessageReaders and MessageBuilders in python, you would have to ask that on the pycapnp repo (I would guess it isn't currently possible but maybe it could be added).
